Content.Server/GameObjects/Components/Damage/DamageOnToolInteractComponent.cs

```diff
@@ -0,0 +1,108 @@
+﻿using Content.Server.GameObjects.Components.Interactable;
+using Content.Server.GameObjects.EntitySystems;
+using Content.Shared.GameObjects.Components.Interactable;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Serialization;
+
+namespace Content.Server.GameObjects.Components.Damage
+{
+    [RegisterComponent]
+    class DamageOnToolInteractComponent : Component, IInteractUsing
+    {
+        #pragma warning disable 649
+        [Dependency] private IEntitySystemManager _entitySystemManager;
+        #pragma warning restore 649
+        public override string Name => "DamageOnToolInteract";
+        private ActSystem _actSystem;
+
+        /* Set in YAML */
+        protected int Damage;
+        protected bool UsingWelding;
+        protected bool UsingAnchoring;
+        protected bool UsingCutting;
+        protected bool UsingMultitool;
+        protected bool UsingPrying;
+        protected bool UsingScrewing;
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+
+            serializer.DataField(ref Damage,            "damage",           0);
+            serializer.DataField(ref UsingWelding,      "usingWelding",      false);
```

Comment:
instead of these 6 fields, one per toolquality, just expose a list of tool qualities that it works on.

Reply:
That seems like a much better implementation then a series of bool checks.

Reply:
I'd prefer your implimentation idea to my own. The system seems more robust for long term use as the Tools list wouldn't be hardcoded.

From my very limitted understanding of the YAML Parser however it would seem to require me to add a serializer to the Enum so the YAML serialiser would be able to know how to define each tool value. Which is something I'm not prepared todo at this point. Especially when I don't think we have any plans to add more tool qualities at this point. 

If anyone else wants to figure out how to it commit. I'll merge it and test it out.

Reply:
You can do something like this:

```cs
//On the class
List<ToolQuality> tools = new List<ToolQuality>();

//In ExposeData
serializer.DataReadFunction("tools", new List<string>(0), parsedtools =>
{
	foreach (var tool in parsedtools)
	{
		tools.add(Enum.Parse(typeof(ToolQuality), tool));
	}
});
```

Reply:
How would that look writen in YAML?
Like this?
`
tool: wrench
tool: crowbar
`

Reply:
```YAML
tools:
- wrenching
- screwing
- welding
```
I believe

Reply:
@RemieRichards 
Thanks for all the help with Parsing YAML.

Reply:
:D 👍